### PR TITLE
Improved spider AI

### LINKED ISSUE #2335: Spiders start attacking in daylight
"Spiders are hostile to the player and to iron golems as long as the light level immediately around them is 9 or less; otherwise they won't attack unless a player attacks first. Hostile spiders will continue to chase the player even if they are exposed to well-lit locations. If a spider sustains damage from a source other than the player, such as falling, its hostility will be reset." - http://minecraft.gamepedia.com/Spider

### PR DESCRIPTION
**Pull request by scribblemaniac**
Fixes #2335

### CODE REVIEW

File: src/Mobs/Spider.cpp [hunk 1]
```diff
@@ -33,3 +34,37 @@ void cSpider::GetDrops(cItems & a_Drops, cEntity * a_Killer)
 
 
 
+
+void cSpider::EventSeePlayer(cEntity * a_Entity)
+{
+	if (!static_cast<cPlayer *>(a_Entity)->IsGameModeCreative() && (this->GetWorld()->GetBlockBlockLight(this->GetPosition()) <= 9))
```
Comment (tigerw) [inline src/Mobs/Spider.cpp]: For consistency with the rest of the code, perhaps remove the `this` keyword?

I suspect that before you get the block light, you will also need to check cWorld::IsChunkLighted and call cWorld::QueueLightChunk and abort if it isn't. You would also have to get the current chunk position through cChunkDef::BlockToChunk(POSX_TOINT, ...).

**Comment by scribblemaniac** [comment]
Implemented proposed changes and retested. Please review.

**Comment by tigerw** [comment]
Completely forgot about GetChunkX/Z(), sorry.

**Comment by scribblemaniac** [comment]
No problem, I got it anyway :wink:

**Pull request merged**